[Fix] Fixes the current package warnings and errors

MohamedRadwan opened the pull request:
#### Changes introduced
<!-- Explain briefly in a small paragraph or in bullet form the changes this PR brings to
     the application -->
- I moved the application back to npm since it doesn't build if there are errors with the dependency tree with is a good saftey net when we are updating packages.
- I updated keycloak-mock so we won't have any vulnerabilities https://github.com/MohamedRadwan/keycloak-mock/releases/tag/1.0
- This should make jenkins run slightly faster

#### Related issue(s)
<!-- If this PR fixes/closes an issue, please prepend that issue number with one of the github
     closing keywords (ex: `fixes`, `closes`, ...) -->
closes #1113

#### Screenshots (if applicable)
<!-- If you have made UI changes to the application, include a screenshot and if the change 
     involves movement, include a GIF. If the UI changes when the application is in mobile view, 
     show a mobile screenshot too. -->


#### Checklist
If the database has been modified:
- [ ] Update database diagram <!-- Updated diagram located in the backend, at `./src/docs/I-Talent database.xml`, with draw.io and updated the png image at `./src/docs/I-Talent database.png` -->
- [ ] Create new migration <!-- Ran `yarn migrate:create` in backend docker container -->

If API endpoints has been modified:
- [ ] Update backend documentation <!-- Updated corresponding swagger documentation in the routers -->
- [ ] Create/Update validators for the API endpoints <!-- Restrict and sanitize user input in the routers with express-validator.github.io -->
- [ ] The API endpoints are properly secured with keycloak 
<!-- Use the `keycloak.protect(roleName)` express middleware in the routes -->

<!-- 
Optional for now, since tests are not working correctly
- [ ] Create tests for your changes
- [ ] Make sure the tests are passing 
-->

If the UI has been modified:
- [ ] It is translated in both language (with no hard coded text) <!-- To sort the keys and remove unused keys in the translation files, run `yarn i18n:cleanup` -->
- [ ] Has been tested on IE
- [ ] The modifications are tab friendly
- [ ] It is accessible

dynamic11 commented on services/frontend/package.json:
```diff
@@ -74,5 +59,24 @@
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "lodash-webpack-plugin": "^0.11.6",
+    "prettier": "^2.2.1",
+    "webpack": "^5.12.1",
```
I don't see webpack as a dep before this change. why do we need it now?

mo-5 replied:
We needed it before but yarn does seem to display the errors. When I moved to npm it showed me the error. It's a requirement by the lodash bundle package that shrinks the size of lodash.

dynamic11 reviewed (commented):
I don't exactly understand why were switching back to npm vs yarn. can you please explainit?

MohamedRadwan commented:
Yeah so what I found is that with npm when you run with compatibility issues between packages it stops the installation. So if we have 2 packages share a package in between for example mom makes sure they share a package with the same version. However with yarn when you run into dependency tree related errors it still installs all the packages. For example if you go into our frontend right now and try npm install it will give you a dependency tree errors and that we need to downgrade some of our packages. For example we updated to react 17 but it's not yet supported by other packages we use.

dynamic11 commented:
ok that makes sense. so are we going to migrate everything over to npm and drop yarn altogether? i dont want to create a mix of the two. Also on another note...a few months ago benoit and i had a discussion about the fact that we should be change our `yarn install` command since we want it to only install based on the lockfile and not the package.json for the purposes of consistency. Please see picture to see what I mean. please use this as inspiration to update this PR to do the same.

![image](https://user-images.githubusercontent.com/11470442/104079619-0d2e3680-51f2-11eb-8031-abe3916a8ffa.png)

MohamedRadwan commented:
I could move it back to yarn since I already changed all the compatible versions together. Yeah in the end we will just use either yarn or npm we cant use the two. I don't mind either one. I could change it back to yarn and just keep the package versions the same.

dynamic11 commented:
@MohamedRadwan I will leave the decision to you. you seem to understand this better than I do

MohamedRadwan commented:
This should be ready for review now. I moved back to yarn since it has parallel download for packages. I updated keycloak mock and made a release for it and added it to the package.json. all our current versions should be compatible.

MohamedRadwan commented:
This should be ready for review now. I moved back to yarn since it has parallel download for packages. I updated keycloak mock and made a release for it and added it to the package.json. all our current versions should be compatible.

MohamedRadwan commented:
This pr is good for review
